### PR TITLE
add cache control middleware and remove useless Response

### LINKED ISSUE #43: add generic JSON response class
It looks like most routes which don't implement `ImgResponse` are manually setting cache-ttl response headers:

image

All of these routes (I think) are JSON responses, and should implement a similar `JSONResponse` object which automatically sets cache-control rather than defining within the scope of each request.  The `JSONResponse` object would be very similar to `ImgResponse`, but of course implementing a different protocol.

### PR DESCRIPTION
**Pull request by vincentsarago**
overtake #59 and closes #43 

cc @geospatial-jeff

### CODE REVIEW

File: tests/routes/test_cog.py [hunk 1]
```diff
@@ -326,3 +329,5 @@ def test_tile_outside_bounds_error(reader, app):
     reader.side_effect = mock_reader
     response = app.get("/cog/tiles/15/0/0?url=https://myurl.com/cog.tif&rescale=0,1000")
     assert response.status_code == 404
+    # NOT THIS MIGHT CHANGE
+    assert response.headers["Cache-Control"] == "public, max-age=3600"
```
Comment (vincentsarago) [inline tests/routes/test_cog.py:333]: Should we add `Cache-Control` to responses with 4** and 5** status code?

File: titiler/endpoints/cog.py [hunk 2]
```diff
@@ -109,7 +101,7 @@ async def cog_metadata(
             "description": "Return an image.",
         }
     },
-    "response_class": ImgResponse,
+    "response_class": Response,
```
Comment (vincentsarago) [inline titiler/endpoints/cog.py:104]: There is no more need of the ImgResponse class (it was setting the cache control by default)

File: tests/routes/test_cog.py (hunk 1 above)
Comment (geospatial-jeff) [reply tests/routes/test_cog.py:333]: I think it makes sense to include for 4XX but not 5XX

Comment (kylebarron) [reply tests/routes/test_cog.py:333]: 5XX is server error right? Definitely shouldn't cache those responses...

Comment (vincentsarago) [reply tests/routes/test_cog.py:333]: cool, gonna skip cache for those

**Pull request merged**